[20.1] SmallRye GraphQL fails on Quarkus 1.7.5.Final

## Description
After https://github.com/graalvm/mandrel/pull/142 SmallRye GraphQL fails on branch 20.1 with:

```
Fatal error:org.graalvm.compiler.debug.GraalError: Missing classes when looking up default methods in io.smallrye.graphql.spi.MetricsService. Consider adding them to the classpath.
	at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.declaresDefaultMethods(ClassInitializationFeature.java:401)
	at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.processInterfaces(ConfigurableClassInitialization.java:606)
	at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.computeInitKindAndMaybeInitializeClass(ConfigurableClassInitialization.java:565)
	at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.computeInitKindAndMaybeInitializeClass(ConfigurableClassInitialization.java:132)
	at com.oracle.svm.hosted.classinitialization.ConfigurableClassInitialization.shouldInitializeAtRuntime(ConfigurableClassInitialization.java:155)
	at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.checkImageHeapInstance(ClassInitializationFeature.java:167)
	at com.oracle.graal.pointsto.meta.AnalysisUniverse.replaceObject(AnalysisUniverse.java:534)
	at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.replaceObject(AnalysisConstantReflectionProvider.java:217)
	at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.interceptValue(AnalysisConstantReflectionProvider.java:188)
	at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readValue(AnalysisConstantReflectionProvider.java:102)
	at com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readFieldValue(AnalysisConstantReflectionProvider.java:78)
	at com.oracle.graal.pointsto.ObjectScanner.scanField(ObjectScanner.java:182)
	at com.oracle.graal.pointsto.ObjectScanner.doScan(ObjectScanner.java:379)
	at com.oracle.graal.pointsto.ObjectScanner.access$500(ObjectScanner.java:61)
	at com.oracle.graal.pointsto.ObjectScanner$3$1.run(ObjectScanner.java:450)
	at com.oracle.graal.pointsto.util.CompletionExecutor.lambda$execute$0(CompletionExecutor.java:173)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NoClassDefFoundError: org/eclipse/microprofile/metrics/MetricRegistry$Type
	at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3166)
	at java.base/java.lang.Class.getDeclaredMethods(Class.java:2309)
	at jdk.internal.vm.ci/jdk.vm.ci.hotspot.HotSpotResolvedObjectTypeImpl.getDeclaredMethods(HotSpotResolvedObjectTypeImpl.java:932)
	at com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.declaresDefaultMethods(ClassInitializationFeature.java:389)
	... 21 more
Caused by: java.lang.ClassNotFoundException: org.eclipse.microprofile.metrics.MetricRegistry$Type
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:471)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 26 more
```

https://github.com/graalvm/mandrel/pull/143/checks?check_run_id=1211202346#step:17:189

## How To Reproduce

```bash
cd /tmp
git clone https://github.com/graalvm/mx
git clone https://github.com/graalvm/mandrel --branch 20.1
git clone https://github.com/graalvm/mandrel-packaging
$JAVA_HOME/bin/java -ea mandrel-packaging/build.java --mx-home /tmp/mx --mandrel-repo /tmp/mandrel --mandrel-home /tmp/mandrelJDK
wget https://github.com/quarkusio/quarkus/archive/1.7.5.Final.tar.gz
tar xf 1.7.5.Final.tar.gz
cd quarkus-1.7.5.Final
export JAVA_HOME=/tmp/mandrelJDK
./mvnw -Dnative -pl integration-tests/smallrye-graphql verify

### Configuration:
 - OS: Ubuntu 18.04
 - Architecture: AMD64
 - Mandrel version: 20.1.x.x
 - JDK version: OpenJDK 11.0.8+10

--- a/substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/classinitialization/ClassInitializationFeature.java
+++ b/substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/classinitialization/ClassInitializationFeature.java
@@ -394,7 +394,7 @@ public class ClassInitializationFeature implements Feature {
             }
             return false;
         } catch (NoClassDefFoundError e) {
-            if (!NativeImageOptions.AllowIncompleteClasspath.getValue()) {
+            if (NativeImageOptions.AllowIncompleteClasspath.getValue()) {
                 return false;
             }
 

